yarn-project/stdlib/src/proofs/proof.ts

```diff
@@ -62,19 +62,43 @@ export class Proof {
   }
 
   public withoutPublicInputs(): Buffer {
-    return Buffer.concat([this.buffer.subarray(this.metadataOffset + Fr.SIZE_IN_BYTES * this.numPublicInputs)]);
+    return Buffer.concat([
+      this.buffer.subarray(this.metadataOffset + Fr.SIZE_IN_BYTES * this.numPublicInputs + this.metadataOffset),
+    ]);
   }
 
+  // This function assumes that the proof will contain an aggregation object
   public extractPublicInputs(): Fr[] {
+    if (this.isEmpty()) {
+      // return array of this.numPublicInputs 0s
+      return new Array(this.numPublicInputs).fill(Fr.zero());
+    }
     const reader = BufferReader.asReader(
-      this.buffer.subarray(this.metadataOffset, this.metadataOffset + Fr.SIZE_IN_BYTES * this.numPublicInputs),
+      this.buffer.subarray(
+        this.metadataOffset,
+        this.metadataOffset + Fr.SIZE_IN_BYTES * (this.numPublicInputs - AGGREGATION_OBJECT_LENGTH),
```

Comment:
please create constants for sizes that are used several times and asserts if possible